main/src/main/scala/sbt/Defaults.scala

```diff
@@ -517,9 +524,9 @@ object Defaults extends BuildCommon {
     ))
 
   lazy val projectTasks: Seq[Setting[_]] = Seq(
-    cleanFiles := cleanFilesTask.value,
+    cleanFiles := Def.taskDyn { cleanFilesTask(managedDirectory.value, target.value) }.value,
```

Comment:
This is the same as it was before. But I'm not sure if I should add `managedDirectory` to `cleanFiles` in the `compile` task above.